File: Java/src/main/java/org/codecop/lcdnumbers/LcdDisplay.java
package org.codecop.lcdnumbers;

import java.util.List;
import java.util.Objects;

/**
 * Combines Converter, Scaler and Printer.
 */
public class LcdDisplay {

    private final DigitsSplitter digitsSplitter;
    private final DigitScaler digitScaler;
    private final DigitPrinter digitPrinter;

    public LcdDisplay(DigitsSplitter digitsSplitter, DigitScaler digitScaler, DigitPrinter digitPrinter) {
        Objects.requireNonNull(digitsSplitter);
        Objects.requireNonNull(digitScaler);
        Objects.requireNonNull(digitPrinter);

        this.digitsSplitter = digitsSplitter;
        this.digitScaler = digitScaler;
        this.digitPrinter = digitPrinter;
    }

    public String toLcd(int number, Scaling scaling) {
        Objects.requireNonNull(scaling);

        List<Digit> digits = digitsSplitter.convert(number);
        List<Digit> scaled = digitScaler.scale(digits, scaling);
        return digitPrinter.render(scaled);
    }
}

File: Csharp-NAutowired/LcdNumbers/LcdDisplay.cs
using System;
using System.Collections.Generic;

namespace Org.Codecop.Lcdnumbers
{
    /// <summary>Combines Converter, Scaler and Printer.</summary>
    public class LcdDisplay
    {
        private readonly DigitsSplitter digitsSplitter;
        private readonly DigitScaler digitScaler;
        private readonly DigitPrinter digitPrinter;

        public LcdDisplay(DigitsSplitter digitsSplitter, DigitScaler digitScaler, DigitPrinter digitPrinter)
        {
            if (digitsSplitter == null)
            {
                throw new ArgumentNullException(nameof(digitsSplitter));
            }
            if (digitScaler == null)
            {
                throw new ArgumentNullException(nameof(digitScaler));
            }
            if (digitPrinter == null)
            {
                throw new ArgumentNullException(nameof(digitPrinter));
            }

            this.digitsSplitter = digitsSplitter;
            this.digitScaler = digitScaler;
            this.digitPrinter = digitPrinter;
        }

        public virtual string ToLcd(int number, Scaling scaling)
        {
            if (scaling == null)
            {
                throw new ArgumentNullException(nameof(scaling));
            }

            IList<Digit> digits = digitsSplitter.Convert(number);
            IList<Digit> scaled = digitScaler.Scale(digits, scaling);
            return digitPrinter.Render(scaled);
        }
    }
}
